ethereum-bridge-contracts/contracts/InboundChannel.sol

```diff
@@ -108,15 +111,16 @@ contract InboundChannel is AccessControl, ISimplifiedMMRProof, ReentrancyGuard {
 
         uint256 results = processMessages(payable(msg.sender), batch.messages);
 
-        uint256 gas_used = block.gaslimit - gasleft();
-        gas_proofs[batch_nonce] = keccak256(abi.encodePacked(gas_used, block.basefee));
+        uint256 gas_used = begin_gas_left - gasleft();
+        gas_proofs[batch_nonce] = keccak256(abi.encode(gas_used, block.basefee));
```

Comment:
I meant add `results`, `batch.messages.length` and `msg.sender` to `gas_proofs` storage :) Not `gas_proofs[batch_nonce]` to event